src/db/transaction.rs

```diff
@@ -77,12 +81,17 @@ impl DbTransactionPool {
         }
     }
 
+    pub fn get_pool(&self) -> Result<Box<dyn DbPool>, Error> {
+        Ok(self.pool.clone())
+    }
+
     /// Perform an action inside of a DB transaction.
-    pub async fn transaction<'a, A, R, F>(&'a self, action: A) -> Result<R, Error>
+    pub async fn transaction<'a, A: 'a, R, F>(&'a self, action: A) -> Result<R, Error>
     where
         A: FnOnce(Box<dyn Db<'a>>) -> F,
         F: Future<Output = Result<R, Error>> + 'a,
     {
+        // Get connection from pool
```

Comment:
nit: This comment is now out of place (and below, in `transaction_http`).

Reply:
https://github.com/mozilla-services/syncstorage-rs/issues/727